lib/checklist_syntax_migrator.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class ChecklistSyntaxMigrator
+  REGEX = /^ {0,3}\[(_|-|\*|\\\*)\]/
+  OLD_CHARACTERS = /(_|-|\*|\\\*)/
+
+  def initialize(post)
+    @post = post
+  end
+
+  def update_syntax!
+    lines =  @post.raw.split("\n")
+    lines.each_with_index do |line, index|
+      lines[index] = line.gsub(REGEX) {
+        Regexp.last_match(0).gsub(OLD_CHARACTERS, "x")
+      }
+    end
+    new_raw = lines.join("\n")
+    @post.raw = new_raw
+    @post.save
+  end
```

Comment:
Could `update_syntax!` be just this instead?

```suggestion
  def update_syntax!
    @post.raw = @post.raw.gsub(/^({,3})\[(_|-|\*|\\\*)\]/) { "#{$1}[x]" }
    @post.save!
    end
  end
```